dev/PowerNotifications/PowerNotifications.cpp

```diff
@@ -0,0 +1,501 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include <pch.h>
+#include <PowerNotifications.h>
+#include <PowerNotificationsPal.h>
+#include <PowerManager.g.cpp>
+
+namespace winrt::Microsoft::ProjectReunion::implementation
+{
+    
+    PowerManager::eventType PowerManager::m_energySaverStatusChangedEvent;
+    PowerManager::eventType PowerManager::m_batteryStatusChangedEvent;
+    PowerManager::eventType PowerManager::m_powerSupplyStatusChangedEvent;
+    PowerManager::eventType PowerManager::m_remainingChargePercentChangedEvent;
+    PowerManager::eventType PowerManager::m_remainingDischargeTimeChangedEvent;
+
+    PowerManager::eventType PowerManager::m_powerSourceStatusChangedEvent;
```

Comment:
Hey @kennykerr - this PowerManager type is mostly static members. What's the C++/WinRT idiomatic way to implement it so we don't have to declare the static-members and then put their definitions here?

Reply:
@jonwis those definitions should not be necessary. Note that static events should be implemented on the factory object with the static_lifetime marker. [Here's an example](https://github.com/microsoft/cppwinrt/blob/198a8270d16cead25edec4a63d8792b267539fd5/test/old_tests/Component/Static.h). I thought I had an article explaining this but I can't find it - @oldnewthing did you ever write an article about static events and factories?

Reply:
Basically, you need to pin the factory to ensure that its lifetime is stable as event handlers are added and removed. That's what `static_lifetime` is for. And you can do it all in a header.

Reply:
I see we have docs for `static_lifetime` - perhaps [this doc](https://docs.microsoft.com/en-us/uwp/cpp-ref-for-winrt/static-lifetime) just needs to be updated to show an example of a static event [like this](https://github.com/microsoft/cppwinrt/blob/198a8270d16cead25edec4a63d8792b267539fd5/test/old_tests/Component/Static.h). @stevewhims is that something you can do?

Reply:
Yes, I've added a second example to the topic: https://docs.microsoft.com/uwp/cpp-ref-for-winrt/static-lifetime.

Reply:
Thanks!